### ISSUE TITLE
Request header cannot handle duplicate keys with different case

### ISSUE THREAD

**Issue by lawrencefinn** [playframework/playframework#4060]
Example:
GET /node HTTP/1.1
User-Agent: curl/7.30.0
Host: localhost:9090
Accept: */*
x-forwarded-for: a, b
X-Forwarded-For: c, d

request.headers.getAll("x-forwarded-for") will only return one of those sets (probably the lowercase one) and won't let you access the other set.

The reason that we have both headers with the same key but different case is because F5 just tags on the header and doesn't look to see what exists.  This sort of call is handled fine by apache.

**Comment by felixmulder** [comment]
Can confirm for play 2.3.8 with using curl:

```bash
$ curl --header "x-forwarded-for: a, b" --header "X-forwarded-for: c, d" localhost:9000
```

```scala
object Application extends Controller {
  def index = Action { request =>
    Ok(request.headers.getAll("x-forwarded-for").mkString("\n"))
  }
}
```

curl receives: "a, b"

**Comment by benmccann** [comment]
See https://github.com/playframework/playframework/pull/3278

**Comment by lawrencefinn** [comment]
Sorry for my ignorance, this is my first time reporting a bug to play or trying to contribute.  It looks like the original pull request didn't pass tests.  Should i try to implement it myself in a pull request and write tests?

**Comment by benmccann** [comment]
@lawrencefinn that sounds great to me. thanks!

**Comment by lawrencefinn** [comment]
What's the prefered solution here?  I can modify the original "pairs" object that is a sequence to have the right values mixed together, i can change that object to a map, or i can fix the toMap function in the headers api to build a correct map

**Comment by benmccann** [comment]
cc @pauldraper @gmethvin @cchantep @jroper  you guys were all active in discussing this over on https://github.com/playframework/playframework/issues/3278 so perhaps you have thoughts?

**Comment by jroper** [comment]
I was happy with @pauldraper's solution, I just want to see tests because something like that is definitely going to regress as we refactor things if there's no tests.

**Comment by lawrencefinn** [comment]
I can re-implement his solution and add tests.

**Comment by pauldraper** [comment]
Should be fixed on master now, due to #3278.
